src/main/java/org/ecocean/Setting.java

```diff
@@ -222,15 +222,20 @@ public static void initialize(String context) {
         Shepherd myShepherd = new Shepherd(context);
         myShepherd.setAction("Setting.initialize");
         myShepherd.beginDBTransaction();
-        Setting st = myShepherd.getSetting("language", "available");
-        if (st == null) {
-            st = new Setting("language", "available");
-            List<String> langs = Arrays.asList(new String[]{"de", "en", "es", "fr", "it"});
-            st.setValue(langs);
-            myShepherd.storeSetting(st);
+        try {
+            Setting st = myShepherd.getSetting("language", "available");
+            if (st == null) {
+                st = new Setting("language", "available");
+                List<String> langs = Arrays.asList(new String[]{"de", "en", "es", "fr", "it"});
+                st.setValue(langs);
+                myShepherd.storeSetting(st);
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        } finally {
+            myShepherd.commitDBTransaction();
```

Comment:
@naknomum On an exception, would you still want to commit here? Could it leave anything in an interim state but now persisted?